File: 백준/Silver/15821. 낚이고 낚아라/낚이고 낚아라.py
def distance(x, y):
    return x**2 + y**2


n, k = map(int, input().split())
v = []
for _ in range(n):
    pi = int(input())
    points = list(map(int, input().split()))
    furthest = max(distance(points[i * 2], points[i * 2 + 1]) for i in range(pi))
    v.append(furthest)
v.sort()

print(f"{v[k - 1]}.00")

File: 백준/Silver/15821. 낚이고 낚아라/낚이고 낚아라.cc
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define MOD 1000000007
#define MAX 100000
#define INF 987654321

ll distance(ll &x, ll &y)
{
    return x * x + y * y;
}
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    ll n, k;
    cin >> n >> k;
    vector<ll> v;
    for (ll i = 0; i < n; i++)
    {
        ll pi;
        cin >> pi;

        ll furthest = 0;
        for (ll i = 0; i < pi; i++)
        {
            ll a, b;
            cin >> a >> b;
            furthest = max(furthest, distance(a, b));
        }
        v.push_back(furthest);
    }

    sort(v.begin(), v.end());

    cout << v[k - 1] << ".00";

    return 0;
}